### DISCUSSION TITLE
GLB Model animations

### DISCUSSION THREAD

**Discussion by AqeelAaqi** [sceneview/sceneview#213] [Q&A]
Hi [ThomasGorisse](https://github.com/ThomasGorisse) I want to change the animation on click and everything is working fine but when I reached the last index I want to move again 0 indexes but this is not working can you please let me know why this is not working. check my code is given below and thanks in advance.

index = 0
btn_animation.setOnClickListener {
++index;
if(index <= animationCount)
{
if(index == animationCount)
{
index = 0
}
playAnimation(index,true)
}
}
playAnimation()

**Comment by ThomasGorisse** [comment]
GLB animations work out of the box in v4.0.0-rc.1. Load with `rememberModelInstance` and control via the animator:

```kotlin
val model = rememberModelInstance(modelLoader, "models/animated.glb")
if (model != null) {
    model.animator?.playAnimation(0) // play first animation
    ModelNode(modelInstance = model)
}
```

Skeletal, morph target, and transform animations are all supported. See `samples/android-demo` for animation demos.

**Comment by ThomasGorisse** [comment]
GLB animations are supported in v4. When you load a model with animations, they are available through `ModelNode`:

```kotlin
val model = rememberModelInstance(modelLoader, "models/animated.glb")

SceneView(/* ... */) {
    if (model != null) {
        ModelNode(
            modelInstance = model,
            autoAnimate = true  // plays all animations automatically
        )
    }
}
```

For manual control, use the `Animator` from the `ModelInstance`:
```kotlin
model?.animator?.let { animator ->
    animator.applyAnimation(0)  // play animation at index 0
    animator.updateBoneMatrices()
}
```

Make sure your GLB exports include the animations — test with [gltf-viewer](https://gltf-viewer.donmccurdy.com/) to verify.
